sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuApproximatePercentile.scala

```diff
@@ -0,0 +1,147 @@
+  override lazy val initialValues: Seq[GpuExpression] = throw new UnsupportedOperationException
+
+  // the update expression will create a t-digest (List[Struct[[Double, Double]])
+  override lazy val updateExpressions: Seq[Expression] =
+    new CudfTDigest(inputBuf,
+      percentageExpression,
+      accuracyExpression) :: Nil
+
+  // the merge expression will merge t-digests
+  override lazy val mergeExpressions: Seq[GpuExpression] =
+    new CudfTDigest(outputBuf,
+      percentageExpression,
+      accuracyExpression) :: Nil
+
+  // the evaluate expression will compute percentiles based on a t-digest
+  override lazy val evaluateExpression: Expression = {
+    ApproxPercentileFromTDigestExpr(outputBuf, percentiles, child.dataType)
+  }
+
+  // inputBuf represents the initial aggregation buffer
+  protected final lazy val inputBuf: AttributeReference =
+    AttributeReference("inputBuf", CudfTDigest.dataType)()
+
+  // outputBuf represents the merged aggregation buffer
+  protected final lazy val outputBuf: AttributeReference =
+    inputBuf.copy("outputBuf", CudfTDigest.dataType)(inputBuf.exprId, inputBuf.qualifier)
+
+  /**
+   * Using child references, define the shape of the vectors sent to the window operations
+   */
+  override val windowInputProjection: Seq[Expression] = Seq.empty
+
+  /**
+   * Create the aggregation operation to perform for Windowing. The input to this method
+   * is a sequence of (index, ColumnVector) that corresponds one to one with what was
+   * returned by [[windowInputProjection]].  The index is the index into the Table for the
+   * corresponding ColumnVector. Some aggregations need extra values.
+   */
+  override def windowAggregation(inputs: Seq[(ColumnVector, Int)]): RollingAggregationOnColumn = {
+    throw new UnsupportedOperationException()
+  }
+
+  override def nullable: Boolean = false
+
+  // Mark as lazy so that percentageExpression is not evaluated during tree transformation.
+  private lazy val (returnPercentileArray, percentiles) =
+    percentageExpression match {
+      case GpuLiteral(v, _) => makeArray(v)
+      case _ => makeArray(percentageExpression.eval())
+    }
+
+  private def makeArray(v: Any)  = v match {
+    // Rule ImplicitTypeCasts can cast other numeric types to double
+    case null => (false, null)
+    case num: Double => (false, Array(num))
+    case arrayData: ArrayData => (true, arrayData.toDoubleArray())
+  }
+
+
+  // The result type is the same as the input type.
+  private lazy val internalDataType: DataType = {
+    if (returnPercentileArray) ArrayType(child.dataType,
+      containsNull = false) else child.dataType
+  }
+
+  override def dataType: DataType = internalDataType
+
+  override def children: Seq[Expression] = Seq(child, percentageExpression, accuracyExpression)
+}
+
+object GpuApproximatePercentile {
+  def apply(child: Expression, percentageExpression: GpuLiteral): GpuApproximatePercentile = {
```

Comment:
If this is not called by the plugin, then lets remove it. Spark adds these types of things to make the SQL translation simpler, but we don't have to deal with that part.